### DISCUSSION TITLE
Which energy statistic sensors would you like to see?

### DISCUSSION THREAD

**Discussion by JensenNick** [JensenNick/huawei_solar_stat#2] [Forum]
Which energy statistics would you like to see in this package? Let me know below and I'll ad them to this list.

### Statistics Sensors under consideration

- [x] **Battery efficiency pr. cycle**<br> Pr. cycle would be pr. last cycle, 7 days, 30 days, 120 days and 365 days. This require data pr. whole cycle. I'm seeing a solution which will require a SQL query and therefore probably be limited to those who use MariaDB or SQL lite.
- [ ] **"Burn Down Chart"** (ROI-chart). The  `sensor.economy_nri_pv` sums up how much you have got in net return, so the information is available, but I would need help to set this up in an Apex Chart.
- [ ] ...
- [ ] ...

### Statistics Sensors already included
- [x] **Self Sufficiency**

| Name | Description |
| -- | -- |
| Energy Self Sufficiency (ts)<br> `energy_self_sufficiency_ts` | -<br> Attributes calculated selfsufficience daily, weekly, monthly, quarterly and yearly. |

- [x] **Yield pr. kWp**

| Name | Description |
| -- | -- |
| Energy Yield Total pr kWp (ts)<br> `energy_yield_total_pr_kwp_ts` | - |
| Energy Yield #1 pr kWp (ts)<br> `energy_yield_1_pr_kwp_ts` | - |
| Energy Yield #2 pr kWp (ts)<br> `energy_yield_2_pr_kwp_ts` | - |

The package includes hourly, daily, weekly, monthly, quarterly and yearly utility meters to track "Yield pr. kWp"

- [x] **Price pr. kWh**

| Name | Description |
| -- | -- |
| Price/kWh wo PV (ts)<br> `price_kwh_wo_pv_ts` | -<br> Attributes include daily, weekly, monthly, quarterly and yearly. |
| Price/kWh w PV (ts)<br> `price_kwh_w_pv_ts` | -<br> Attributes include daily, weekly, monthly, quarterly and yearly. |
| Price/kWh Export (ts)<br> `price_kwh_export_ts` | -<br> Attributes include daily, weekly, monthly, quarterly and yearly. |
| Price/kWh Export Yield (ts)<br> `price_kwh_export_yield_ts` | -<br> Attributes include daily, weekly, monthly, quarterly and yearly. |
| Price/kWh Export Battery Discharge (ts)<br> `price_kwh_export_battery_discharge_ts` | -<br> Attributes include daily, weekly, monthly, quarterly and yearly. |

- [x] **Battery Statistics**

| Name | Description |
| -- | -- |
| Energy Battery Discharge Capacity (ts)<br> `energy_battery_Discharge_capacity_ts` | Available capacity for discharge, with your current settings |
| Enegy Battery Charge Grid Capacity (ts)<br> `energy_battery_charge_grid_capacity_ts` | Available space on your battery for charging over AC. *This could come in handy for charging automations* |
| Enegy Battery Charge Yield Capacity (ts)<br> `energy_battery_charge_yield_capacity_ts` | Available space on your battery for charging yield from your solar PV. *This could come in handy for charging automations* |
| Battery Efficiency Total (tt)<br> `battery_efficiency_total_tt` | The "total" efficiency of your battery, meaning from start of measuring until "now". |

**Comment by AndersNielsen89** [comment]
Would be awesome to have a "burn down" chart of projected payback time of your PV system. Obviously this metric is best when all production and electricity price data is available historically
![image](https://github.com/JensenNick/huawei_solar_stat/assets/4113075/ec35d8cf-705a-457f-a8e1-954c00f71a50)

**Comment by JensenNick** [comment]
> Would be awesome to have a "burn down" chart of projected payback time of your PV system.

@AndersNielsen89 Thank you for your input! I believe you have all the information you need (or at least the information I can help to provide) in the _"Huawei Solar PEES package"_ 🙂. Apart from the information from the sensors you would need the amount of your investment and if it has to be really fancy maybe some financial regulation factor.

The `sensor.economy_nri_pv` is = the savings + earnings you have compared to what you would otherwise have had to pay for the electricity you have consumed. NRI = Net Return of Investment. 

And with regards to the historical data, I hope I've set at `state_class:` on all the relevant data - this makes sure HA saved the data long term (history). Not as detailed as short term data, but every hour should suffice for most use cases I guess. You could consider setting up InfluxDB + Grafana as well.

How to set this up in an Apex Chart or Grafana is another story. That part does not have my focus right now as my priority is to get the right setup an save the right information.
